android/app/src/main/java/org/uwaterloo/subletr/pages/watcardverification/WatcardVerificationPageView.kt

```diff
@@ -0,0 +1,296 @@
+	uiState: WatcardVerificationUiState = viewModel.uiStateStream.subscribeAsState(
+		WatcardVerificationUiState.Loading
+	).value
+) {
+
+	if (uiState is WatcardVerificationUiState.Loading) {
+		Column(
+			modifier = modifier
+				.fillMaxSize(1.0f)
+				.imePadding(),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is WatcardVerificationUiState.Loaded) {
+		Column(
+			modifier = modifier.fillMaxSize(1.0f),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			Spacer(modifier = modifier.weight(6.0f))
+			
+			Text(
+				text = stringResource(id = R.string.watcard_verification),
+				style = MaterialTheme.typography.titleMedium,
+				color = primaryTextColor
+			)
+
+			Column(modifier = modifier
+				.fillMaxWidth(1.0f)
+				.padding(horizontal = dimensionResource(id = R.dimen.l))
+			) {
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt1),
+					color = secondaryTextColor,
+				)
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt2),
+					color = secondaryTextColor,
+				)
+			}
+
+			Spacer(modifier = modifier.weight(2.0f))
+
+			SetContent(
+				viewModel = viewModel,
+				watcard = uiState.watcard
+			)
+
+			Spacer(modifier = modifier.weight(8.0f))
+
+			SetSubmitButton(
+				viewModel,
+				uiState.watcard,
+				uiState.submitted,
+				uiState.verified,
+			)
+
+			Spacer(modifier = modifier.weight(3.0f))
+
+			SetCancelButton(
+				uiState.watcard,
+				uiState.submitted,
+				uiState.verified,
+			)
+
+			Spacer(modifier = modifier.weight(8.0f))
+		}
+	}
+}
+
+@Composable
+fun SetContent(viewModel: WatcardVerificationPageViewModel, watcard : String?) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.l))
+			.fillMaxWidth()
+			.height(dimensionResource(id = R.dimen.watcard_image_height))
+			.border(
+				width = dimensionResource(id = R.dimen.xxxs),
+				color = subletrBlack,
+				shape = RectangleShape,
+			)
+			.background(Color.Transparent)
+			.clickable {
+				viewModel.updateUiState(
+					WatcardVerificationUiState.Loaded(
+						watcard = "uploaded",
+						submitted = false,
+						verified = false,
+					)
+				)
+			},
+		contentAlignment = Alignment.Center,
+	) {
+		if (watcard == null) {
+			Column(
+				horizontalAlignment = Alignment.CenterHorizontally,
+				verticalArrangement = Arrangement.Center,
+			) {
+				Icon(
+					modifier = Modifier.size(dimensionResource(id = R.dimen.xl)),
+					painter = painterResource(
+						id = R.drawable.add_filled_black_32
+					),
+					contentDescription = stringResource(id = R.string.add_sign),
+				)
+				Text(
+					stringResource(id = R.string.add_watcard),
+					style = MaterialTheme.typography.bodyMedium,
+				)
+			}
+		} else {
+			// TODO: show picture
+		}
+	}
+}
+
+@Composable
+fun SetSubmitButton(
```

Comment:
submit button text is black when background is pink, this should be textonsubletrpink:
![image](https://github.com/andr-dev/CS446/assets/45717469/34731bf3-a5d7-440e-81f3-f078e5e49f89)

this is likely because youre setting the button's content colour but instead you should be setting the text colour based on the state of the button